ui/hooks/useScrollRequired.js

```diff
@@ -38,7 +41,11 @@ export const useScrollRequired = (
         offsetPxFromBottom >=
         ref.current.scrollHeight;
 
-    setIsScrollable(isScrollable);
+    if (isScrollable !== isScrollableState) {
```

Comment:
This warrants more explanation, why do we have this condition resetting `hasScrolledToBottom`?

Have we checked this doesn't break other usages of the hook?

Reply:
So far I haven't found that something is broken on this PR after this change, e2e tests are also passing all. I will need some assistance from @GuillaumeRx for explanation on this.

Reply:
This resets the `hasScrolledToBottom` when the the page changes from non-scrollable to scrollable. It avoids the component setting the `hasScrolledToBottom` state to `true` before content has loaded inside it (due to the page being empty, nothing is scrollable) and keeping it to `true` when the content has loaded and the page became scrollable. Our problem was that the permissions would not be present when this hook was initialized so the page would not be scrollable, setting the `hasScrolledToBottom` to `true` and never resetting it to `false` when the page became scrollable due to the permissions appearing on the screen. Now when there's a change in the page scrolling behavior it resets the `hasScrolledToBottom` state.

Reply:
Makes sense!